src/EnemyBehaviourEC.cpp

```diff
@@ -30,6 +32,51 @@ EnemyBehaviourEC::~EnemyBehaviourEC() {
     delete distanceToPlayer;
 }
 
+void EnemyBehaviourEC::destroy() {
+    std::vector<Entity*> enemies = scene->getEntitiesbyTag("Enemy");
+
+    for (auto it : enemies) {
+        if (it != father) {
+            Component* comp = it->findComponent("MeleeEnemyBehaviourEC");
+            if (comp == nullptr)
+                comp = it->getComponent("RangedEnemyBehaviourEC");
+
+            removeTransforms(dynamic_cast<EnemyBehaviourEC*>(comp));
+        }
+    }
+    setActive(false);
+    scene->getComponentsManager()->eraseEC(this);
```

Comment:
I'm pretty sure you can do this:
```suggestion
    EventComponent::destroy();
```

Reply:
No estaba terminado sin más hice el pr para mas adelante